modules/webbrowser/src/eventhandler.cpp

```diff
@@ -192,30 +192,84 @@ void EventHandler::initialize() {
             return false;
         }
     );
-}
 
-void EventHandler::touchPressCallback(const double x, const double y) {
-    if (_browserInstance) {
-        _mousePosition.x = static_cast<float>(x);
-        _mousePosition.y = static_cast<float>(y);
+    global::callback::touchDetected.emplace_back(
+        [&](TouchInput input) -> bool {
+            if(!_browserInstance) return false;
 
-        int clickCount = BrowserInstance::SingleClick;
-        _browserInstance->sendMouseClickEvent(mouseEvent(), MBT_LEFT, false, clickCount);
-    }
-}
+            glm::vec2 windowPos = input.getCurrentWindowCoordinates();
+            if(!_browserInstance->hasContent(static_cast<int>(windowPos.x),
+                                            static_cast<int>(windowPos.y))){
+                return false;
+            }
 
-void EventHandler::touchReleaseCallback(const double x, const double y) {
-    if (_browserInstance) {
-        _mousePosition.x = static_cast<float>(x);
-        _mousePosition.y = static_cast<float>(y);
+            if(_validTouchStates.empty()) {
+                _mousePosition.x = windowPos.x;
+                _mousePosition.y = windowPos.y;
+                _leftButton.down = true;
+                _browserInstance->sendMouseClickEvent(mouseEvent(),
+                                                        MBT_LEFT,
+                                                        false,
+                                                        BrowserInstance::SingleClick);
```

Comment:
The OpenSpace coding style says to just indent arguments on separate lines by 4 spaces when calling a function.